ocaml/xapi/xapi_pbd.ml

```diff
@@ -113,21 +113,20 @@ let check_sharing_constraint ~__context ~sr =
   end
 
 (** If the SR requires some cluster stacks, we resync every compatible Cluster *)
-let resync_cluster_stack_for_sr_type ~__context ~sr_sm_type =
+let resync_cluster_stack_for_sr_type ~__context ~sr_sm_type ~host =
   let required_cluster_stacks = Xapi_clustering.get_required_cluster_stacks ~__context ~sr_sm_type in
   (* This is empty if the SR requires no cluster stack *)
-  let required_clusters =
-    Db.Cluster.get_all_records ~__context
-    |> List.filter (function (cluster_ref, cluster_rec) -> List.mem cluster_rec.API.cluster_cluster_stack required_cluster_stacks)
-  in
-  (* XXX For now, we only support one cluster, when we add support for
-     multiple clusters, we may want to change this behaviour. *)
-  required_clusters
-  |> List.iter
-    (fun (cluster_ref, cluster_rec) ->
-       Helpers.call_api_functions ~__context (fun rpc session_id ->
-           Client.Client.Cluster.pool_resync ~rpc ~session_id ~self:cluster_ref)
+  if List.exists
+  (* check for a cluster_host associated with the host and with a cluster with a matching cluster_stack *)
+    (fun (_, cluster_host_rec) ->
+      let self = cluster_host_rec.API.cluster_host_cluster in
+      List.mem (Db.Cluster.get_cluster_stack ~__context ~self) required_cluster_stacks
     )
+    (Db.Cluster_host.get_records_where ~__context
```

Comment:
Could this be simplified by using Xapi_clustering.find_cluster_host?

Reply:
Yes, that would be smarter than what I did :P I was concerned about unnecessarily repeating code as resync_host calls this, but that's after calling `create_as_necessary` so it makes sense.